source/simulator/assemblers/advection.cc

```diff
@@ -91,6 +91,7 @@ namespace aspect
             {
               if (fe.system_to_component_index(i).first == solution_component)
                 {
+                  scratch.hessian_phi_field[i_advection]        = scratch.finite_element_values[solution_field].hessian (i,q);
```

Comment:
Since we only need this with SUPG, can you move `use_supg` up and only do this if necessary? It will make assembly more expensive.